### PR TITLE
Fix for `theme serve` to trigger page refresh when a file is deleted

### LINKED ISSUE #2342: [Bug]: `shopify theme serve` - file deletions to not trigger hot reload
### Please confirm that you have:

- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

Just like file additions and updates I'd expect a file deletion to trigger to hot reloading process

### Actual behavior

When I delete a file the CLI detects the deletion of it but the page does not refresh

### Stack trace

_No response_

### Reproduction steps

1. Run `shopify theme serve`
2. Remove a file (the file extension does not matter)

### Operating System

macOS Monterey

### CLI version (run `shopify version` if you're not sure)

2.17.0

### Shell

zsh

### Ruby version (run `ruby -v` if you're not sure)

2.6.8p205

### What language and version are you using in your application?

ruby 3.1.2

### PR DESCRIPTION
**Pull request by mgmanzella**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify-cli/issues/2342

### WHAT is this pull request doing?

Added logic in hot_reload.rb to recognize when files are deleted and reload the page after the files have been deleted remotely

#### Setup (if you dont have the dev cli or a theme ready):
- Clone the `shopify/shopify-cli`
- Create an alias to the CLI with `alias shopify-dev='/Shopify/shopify-cli/bin/shopify'`
- Run `shopify-dev theme init test_theme`
- Run `cd shopify-dev`
- Run `shopify-dev login -s <your_store>`

#### Test hot reloading a deleted file
- Run `shopify-dev theme serve `
- (wait for the progress bar)
- Open `http://127.0.0.1:9292` in your browser 
- Navigate to your theme and delete `announcement-bar.liquid`
- Confirm the page reloads and there is an error about the liquid file not being found
Screen Shot 2022-07-11 at 2 29 52 PM

NOTE: re-adding the deleted file will not be picked up by hot reloading, you will have to restart the theme serve command. This bug is captured [here](https://github.com/Shopify/shopify-cli/issues/2447)

For regression testing, make any modifications and new file additions to see they get picked up by the dev server

### Post-release steps

None

### Update checklist

- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).
- [x] I've included any post-release steps in the section above (if needed).

### CODE REVIEW

File: lib/shopify_cli/theme/dev_server/hot_reload.rb [hunk 1]
```diff
@@ -36,13 +38,20 @@ def close
...
+        def notify_streams_of_file_change(modified, added, removed)
           files = (modified + added)
             .reject { |file| @ignore_filter&.ignore?(file) }
             .map { |file| @theme[file] }
 
           files -= liquid_css_files = files.select(&:liquid_css?)
 
+          deleted_files = removed
+            .reject { |file| @ignore_filter&.ignore?(file) }
+            .map { |file| @theme[file] }
```
Comment (jessetuglu) [inline lib/shopify_cli/theme/dev_server/hot_reload.rb:50]: Just a thought — I think the following results in less allocations:
```ruby
filter_map { |file| @theme[file] if !@ignore_filter&.ignore?(file) }
```

**Review by ulgut — COMMENTED** [review]
Just left a few comments, but feel free to ignore 😄 👍

File: lib/shopify_cli/theme/dev_server/hot_reload.rb [hunk 2]
```diff
@@ -55,8 +64,21 @@ def hot_reload(files)
...
         end
 
+        def reload_page(removed)
+          @streams.broadcast(JSON.generate(reload_page: true))
+          @ctx.debug("[ReloadPage] Deleted #{removed.join(", ")}")
+        end
+
+        def remote_delete(files)
+          files.each do |file|
+            puts("delete file each -> file.relative_path #{file.relative_path}")
```
[the remark this answers is not in the document]
Comment (mgmanzella) [reply lib/shopify_cli/theme/dev_server/hot_reload.rb:74]: great call out, yes i think youre right

File: lib/shopify_cli/theme/dev_server/hot_reload.rb (hunk 1 above)
Comment (mgmanzella) [reply lib/shopify_cli/theme/dev_server/hot_reload.rb]: very slick 🔥

**Review by ulgut — APPROVED** [review]
🦐 👍

**Comment by mgmanzella** [comment]
Integration tests and bulk tests seem to fail periodically, the team is investigating but this PR is blocked until we can track down the source of these failing tests

**Review by karreiro — COMMENTED** [review]
Thank you, @mgmanzella!

I've left a suggestion on Slack regarding some debug elements (like the `test.sh`). But the PR looks good, so +1 for merging it after we apply those changes :)

**Review by karreiro — COMMENTED** [review]
Thank you, @mgmanzella! The PR looks great and works fine as well 🎩 I've left only one minor comment related to a JS error that happens when the client gets a server side event.

Please, let me know wdyt :)

**Review by karreiro — APPROVED** [review]
Thank you, @mgmanzella!

**Pull request merged**